packages/peregrine/lib/talons/Header/useCartTrigger.js

```diff
@@ -15,15 +16,17 @@ export const useCartTrigger = props => {
     } = props;
 
     const apolloClient = useApolloClient();
-    const [, { toggleDrawer }] = useAppContext();
     const [{ cartId }, { getCartDetails }] = useCartContext();
+    const history = useHistory();
 
     const [getItemCount, { data }] = useLazyQuery(getItemCountQuery, {
         fetchPolicy: 'cache-and-network'
     });
     const [fetchCartId] = useMutation(createCartMutation);
     const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);
 
+    const [miniCartIsOpen, setMiniCartIsOpen] = useState(false);
```

Comment:
What do we gain from using local state? I think we can still get away with using `toggleDrawer`. You could even just call it `cart-2` for now and in the new mini cart talon we would just check for `drawer === 'cart-2'` instead of threading the `isOpen` prop around.

Reply:
> Basically, I am intentionally trying to bring the state local here - aren't we trying to get away from the site-wide contexts (useAppContext::toggleDrawer)?

We're avoiding global state for things that aren't global state. If the new mocks call for multiple drawers being opened at once then this would no longer fit into the old drawer pattern. So that's fine.